Htww2

abdulredd opened the pull request:
# Curriculum PR Template

### Does this PR close any issues?
#164 

### Description
Some significant changes to the lesson to include an Intro to APIs.

I pulled in some stuff from the Launch lesson, and also brought in some Postman exploration from the Faraday and Postman lesson from B3, mainly working with the `https://pets-api.turingschool-examples.com/api/v1/pets` endpoint.

### What questions do you have/what do you want feedback on? (optional)
I wasn't sure if keeping the `curl` exploration was necessary or not, but I kept it in for now. In how the web works 1, there is an intro to the request/response cycle, but no exploration beyond identify the different components of the request and response.

The major concern I have with this session is that it is only an hour, and I believe there might be too much to cover in that short of a time. All thoughts/suggestions welcome.

epintozzi reviewed (commented):
Based on the revised Scope and Sequence doc, it sounds like we're anticipating allowing up to 3 hours for this lesson. Currently the calendar has it for 2. Do you think 2 is enough? It's scheduled for Friday of week 2.

KatBrandt commented on module2/lessons/index.md:
```diff
@@ -4,45 +4,55 @@ title: Module 2 - Lessons
+- [Chrome Dev Tools](./chrome_dev_tools)
+- [JSON for API development](../resources/json_for_api_development)
+- [How the Web Works I: HTTP Request/Response Cycle](./how_the_web_works_http)
+- [How the Web Works II: APIs](./how_the_web_works_api)
```
Such a small nitpick so feel free to disregard, but I feel like these 2 lessons shouldn't be in "additional resources" to me "additional resources" are nice to knows not need to knows, are things students should be exploring on their own time, etc. I wonder if we made a Web Development section header if these couldn't go in there?

abdulredd replied:
Yea, I looked through the available categories and couldn't decide where this lesson fit, so I stuck it here. Any suggestions are welcome.

KatBrandt commented on module2/lessons/how_the_web_works_api.md:
````diff
@@ -0,0 +1,219 @@
+* Connected to google.com (2607:f8b0:400f:800::200e) port 80 (#0)
+> GET / HTTP/1.1
+> Host: google.com
+> User-Agent: curl/7.43.0
+> Accept: */*
+>
+< HTTP/1.1 301 Moved Permanently
+< Location: http://www.google.com/
+< Content-Type: text/html; charset=UTF-8
+< Date: Thu, 31 Aug 2017 01:09:30 GMT
+< Expires: Sat, 30 Sep 2017 01:09:30 GMT
+< Cache-Control: public, max-age=2592000
+< Server: gws
+< Content-Length: 219
+< X-XSS-Protection: 1; mode=block
+< X-Frame-Options: SAMEORIGIN
+<
+<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
+<TITLE>301 Moved</TITLE></HEAD><BODY>
+<H1>301 Moved</H1>
+The document has moved
+<A HREF="http://www.google.com/">here</A>.
+</BODY></HTML>
+* Connection #0 to host google.com left intact
+```
+
+The `>` symbols indicate part of the request being sent to the server. The `<` symbols indicate the response coming back to our system.
+
+Sending a `curl` request to `https://google.com` will provide the actual site that is displayed when we visit google in a browser.
+
+<section class="checks-for-understanding">
+### Check for Understanding
+
+See if you can identify the different parts of the HTTP request and HTTP response.
+
+</section>
+
+#### curl Student Site
+
+Send a `curl` request to a web site of your choice. Note many of the same pieces from the request/response above, with slightly more HTML. Also note, the CSS is not in any way included in the response when you make a request to a student site (unless they used inline CSS). In order to get a CSS file we need to make a second request to something like `https://username.github.io/main.css`.
````
I think it's fine to keep it, just make sure this doesn't lead into a curl discussion. It's just a tool to make http requests in our terminal, we'll see better tools later in this lesson, postman.

KatBrandt commented on module2/lessons/how_the_web_works_api.md:
```diff
@@ -0,0 +1,219 @@
+When thinking about the role that APIs play in our web applications, they are another data source for us to call on to retrieve some data, similar to accessing data from your own database. The biggest difference being, you don't have to manage the API database yourself. How cool is that!
+
+<section class="call-to-action">
+### What are some potential drawbacks of using an API?
```
I'd love to see this be a drop down with bullet points inside of what some of those drawbacks are. I feel like our lessons should be fully built out so that a student could read them and not be left wondering what the answer is. :) IMO

abdulredd replied:
I was thinking of this more like a turn-and-talk. I could also add some examples here as well.

hfaerber replied:
@KatBrandt 

> our lessons should be fully built out so that a student could read them and not be left wondering what the answer is

I agree with this.  As open source curriculum, I think this is a really valuable way to approach our lessons.

abdulredd commented:
So I updated the callout box to a dropdown and added some responses. I'm still going back and forth whether to leave in the curl exploration, but I left it in for now since I am the one teaching this lesson, I'll see how the lesson goes and see it all shakes out at the end. Merging to main...
